sdk/internal/recording/recording.go

```diff
@@ -524,14 +524,16 @@ var client = http.Client{
 
 type RecordingOptions struct {
 	UseHTTPS        bool
+	ProxyPort       int
 	GroupForReplace string
 	Variables       map[string]interface{}
 	TestInstance    *testing.T
 }
 
 func defaultOptions() *RecordingOptions {
 	return &RecordingOptions{
-		UseHTTPS: true,
+		UseHTTPS:  true,
+		ProxyPort: os.Getpid()%10000 + 20000,
```

Comment:
ohhh this is smart. They're each individual process ids so they _can't_ stomp on each other.

Reply:
There could still be a port conflict that may be worth checking. @scbedd the other option I looked at was connecting to `localhost:0` which makes the OS assign a random port, and then closing the listener and using that as the proxy port, but this code gets called from a few places with a null value for the options, so it was more annoying to refactor everything to propagate the state of the initial port, rather than pick something deterministic like this.

Reply:
Does the test-proxy print out the port when you launch it? Seems like a nice to have for cases like this.

Reply:
Yep! The logs look like this:

> 2023/07/14 20:36:11 Test proxy version 1.0.0-dev.20230706.1 already installed
> 2023/07/14 20:36:11 Running test proxy command: /home/ben/sdk/azure-sdk-for-go/.proxy/Azure.Sdk.Tools.TestProxy start --storage-location /home/ben/sdk/azure-sdk-for-go -- --urls=https://localhost:29432
> 2023/07/14 20:36:11 Test proxy log location: /tmp/testproxy.log.uycawv
> 2023/07/14 20:36:13 Started test proxy instance (PID 19459) on https://localhost:29432
> 2023/07/14 20:36:15 Stopping test proxy instance (PID 19459) on https://localhost:29432